objective.py

```diff
@@ -161,6 +173,8 @@ def get_objective(self):
         sfreq: sampling frequency to allow filtering the data.
         """
 
+        X_train, X_test, y_train, y_test = self.get_split(self.X, self.y)
```

Comment:
Does this work? I don't see how you can get the split here but maybe I am missing something?